reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java

```diff
@@ -617,6 +632,14 @@ else if (channel().pipeline()
 	@Override
 	protected void onInboundNext(ChannelHandlerContext ctx, Object msg) {
 		if (msg instanceof HttpRequest) {
+			if (readTimeout != null) {
+				addHandlerFirst(NettyPipeline.ReadTimeoutHandler,
+						new ReadTimeoutHandler(readTimeout.toMillis(), TimeUnit.MILLISECONDS));
+			}
+			if (requestTimeout != null) {
+				requestTimeoutFuture =
+						ctx.executor().schedule(new RequestTimeoutTask(ctx), Math.max(requestTimeout.toMillis(), 1), TimeUnit.MILLISECONDS);
+			}
```

Comment:
is it possible to avoid scheduling the timers in case we receive a Full H2 or H2C Full Request ? 
if I'm correct, when receiving for example a Full H2/H2C GET request, then the msg may be an instance of **DefaultFullHttpRequest**, meaning that we will first arm timers here:

```
	protected void onInboundNext(ChannelHandlerContext ctx, Object msg) {
		if (msg instanceof HttpRequest) {
			if (readTimeout != null) {
-->				addHandlerFirst(NettyPipeline.ReadTimeoutHandler,
						new ReadTimeoutHandler(readTimeout.toMillis(), TimeUnit.MILLISECONDS));
			}
			if (requestTimeout != null) {
-->				requestTimeoutFuture =
						ctx.executor().schedule(new RequestTimeoutTask(ctx), Math.max(requestTimeout.toMillis(), 1), TimeUnit.MILLISECONDS);
			}
```

and then we will immediately cancel the timers here:

```
	@Override
	protected void onInboundNext(ChannelHandlerContext ctx, Object msg) {
		if (msg instanceof HttpRequest) {
			if (readTimeout != null) {
				addHandlerFirst(NettyPipeline.ReadTimeoutHandler,
						new ReadTimeoutHandler(readTimeout.toMillis(), TimeUnit.MILLISECONDS));
			}
			if (requestTimeout != null) {
				requestTimeoutFuture =
						ctx.executor().schedule(new RequestTimeoutTask(ctx), Math.max(requestTimeout.toMillis(), 1), TimeUnit.MILLISECONDS);
			}
			try {
				listener().onStateChange(this, HttpServerState.REQUEST_RECEIVED);
			}
			catch (Exception e) {
				onInboundError(e);
				ReferenceCountUtil.release(msg);
				return;
			}
			if (msg instanceof FullHttpRequest) {
				FullHttpRequest request = (FullHttpRequest) msg;
				if (request.content().readableBytes() > 0) {
					super.onInboundNext(ctx, msg);
				}
				else {
					request.release();
				}
				if (isHttp2()) {
-->					removeHandler(NettyPipeline.ReadTimeoutHandler);
					if (requestTimeoutFuture != null) {
-->						requestTimeoutFuture.cancel(false);
						requestTimeoutFuture = null;
					}
```

maybe we could avoid scheduling and cancelling the timers in this case ?

Reply:
`FullHttpRequest` is when we receive `Http2HeadersFrame` with `endStream=true`.
I'll fix it. Thanks.